inc/fusioninventory.class.php

```diff
@@ -0,0 +1,198 @@
+<?php
+/**
+ * LICENSE
+ *
+ * Copyright © 2016-2018 Teclib'
+ * Copyright © 2010-2018 by the FusionInventory Development Team.
+ *
+ * This file is part of Flyve MDM Plugin for GLPI.
+ *
+ * Flyve MDM Plugin for GLPI is a subproject of Flyve MDM. Flyve MDM is a mobile
+ * device management software.
+ *
+ * Flyve MDM Plugin for GLPI is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * Flyve MDM Plugin for GLPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Flyve MDM Plugin for GLPI. If not, see http://www.gnu.org/licenses/.
+ * ------------------------------------------------------------------------------
+ * @author    Thierry Bugier
+ * @copyright Copyright © 2018 Teclib
+ * @license   AGPLv3+ http://www.gnu.org/licenses/agpl.txt
+ * @link      https://github.com/flyve-mdm/glpi-plugin
+ * @link      https://flyve-mdm.com/
+ * ------------------------------------------------------------------------------
+ */
+
+ use GlpiPlugin\Flyvemdm\Exception\FusionInventoryRuleInconsistency;
+
+if (!defined('GLPI_ROOT')) {
+   die("Sorry. You can't access this file directly");
+}
+
+class PluginFlyvemdmFusionInventory {
+
+   const RULE_NAME = 'Flyve MDM invitation to entity';
+
+   /**
+    * Creates or updates an entity rule
+    *
+    * @param PluginFlyvemdmInvitation $invitation invitation to handle
+    */
+   public function addInvitationRule(PluginFlyvemdmInvitation $invitation) {
+      $entityId = $invitation->getField(Entity::getForeignKeyField());
+      $rule = $this->getRule($entityId);
+
+      $ruleCriteria = new RuleCriteria();
```

Comment:
I have a question for this function

Don't you think it's better for performance to just compare if the rule exist and is active to then do a return?

If the rule is not active but exist, update it and if it not exist just add it

Reply:
A rule is combination of 3 different itemtypes:
- rule itself
- one or several criterias (with a boolean connector AND or OR)
- one or several actions

As I need to manupulate several itemtypes with relatios between them I choose to implement a class with this specific logic. The invitation class is already quite big.

I had 2 choices : 
- one rule per invitation with a single criteria
- one rule with a lot of criterias, one criteria per invitation

I choosed the 2nd, w th a boolean connector OR